### PR TITLE
ci: test on MacOS

### PR DESCRIPTION
**Pull request by lars-reimann**
### Summary of Changes

* Run tests on MacOS.

### CODE REVIEW

**Comment by lars-reimann** [comment]
@Marsmaennchen221 Should we skip the failing tests on MacOS or is there a way to guarantee consistent behavior across platforms?

**Comment by Marsmaennchen221** [comment]
@lars-reimann do you have a way to create snapshots on macos? Then we could generate snapshots for each os

File: tests/helpers/_operating_systems.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import platform
+import pytest
+
+os_windows = "Windows"
+os_linux = "Linux"
+os_mac = "Darwin"
+
+
+def skip_os_dependent(os: str | list[str]) -> None:
```
Comment (lars-reimann) [inline tests/helpers/_operating_systems.py:10]: I'd find the negation more intuitive: `_skip_if_os`

**Comment by lars-reimann** [comment]
> @lars-reimann do you have a way to create snapshots on macos? Then we could generate snapshots for each os

Only in CI, but that's too cumbersome.

**Pull request merged**